custom/icds_reports/sqldata/exports/dashboard_usage.py

```diff
@@ -0,0 +1,209 @@
+            location_id = location_type + '_id'
+        return location_id
+
+    def get_location_type_string_from_location_id(self, location_id):
+        return location_id.replace('_id', '')
+
+    def check_if_date_in_last_week(self, date):
+        if date is None:
+            return False
+        d = datetime.datetime.strptime(date, "%Y-%m-%d")
+        now = datetime.datetime.now()
+        return (d - now).days < 7
+
+    def prepare_is_launched_agg_list(self, location_type, location_id):
+        '''
+        populates the location_id vs is_launched status and location_id vs location_type mappings for the
+         descendants of given location
+        :param location_type:
+        :param location_id:
+        :return: None
+        '''
+        location_type = self.get_location_id_string_from_location_type(location_type)
+        sub_location_types = self.agg_required_fields[self.location_types.index(location_type) + 1:]
+        filter_dict = {location_type: location_id}
+        aggregate_records = AggAwc.objects.filter(**filter_dict).values(*sub_location_types)
+        aggregate_records_dict = defaultdict(str)
+        location_type_id_mapping = defaultdict(str)
+        for aggregate_record in aggregate_records:
+            for sub_location_type in sub_location_types:
+                if sub_location_type != 'is_launched':
+                    if aggregate_record[sub_location_type] not in aggregate_records_dict:
+                        aggregate_records_dict[aggregate_record[sub_location_type]] =\
+                            aggregate_record['is_launched']
+                    if aggregate_record[sub_location_type] not in location_type_id_mapping:
+                        location_type_id_mapping[aggregate_record[sub_location_type]] = sub_location_type
+        self.agg_list = aggregate_records_dict
+        self.sql_locations = location_type_id_mapping
+
+    def get_excel_data(self):
+        excel_rows = []
+        filters = [['Generated at', india_now()]]
+        headers = ['Sr.No', 'State/UT Name', 'District Name', 'Block Name', 'Username', 'Level', 'Role',
+                   'Launched?', 'Last Login', 'Logged in the last week?', 'Total', 'Child', 'Pregnant Women',
+                   'Demographics', 'System Usage', 'AWC infrastructure', 'Child Growth Monitoring List',
+                   'ICDS - CAS Monthly Register', 'AWW Performance Report', 'LS Performance Report',
+                   'Take Home Ration']
+        excel_rows.append(headers)
+        serial_count = 0
+        for user_location in self.user.get_sql_locations(self.domain):
+            # getting the location types to retrieve for this user location
+            location_type_filter = {
+                self.get_location_id_string_from_location_type(user_location.location_type_name):
+                    user_location.get_id}
+            all_awc_locations = AwcLocation.objects.filter(**location_type_filter).values(*self.required_fields)
```

Comment:
you would have to use the aggregation_level=5 else you will be lots of duplicate rows with entries as 'all' for lower locations.

Reply:
added